src/workflows/steps.js

```diff
@@ -0,0 +1,404 @@
+		state.branch = branch.trim();
+	} );
+}
+
+export function gitFetchUpstreamMaster( state ) {
+	return git.fetchUpstreamMaster();
+}
+
+export function gitMergeUpstreamBranch( state ) {
+	const { branch } = state;
+	return git.merge( `upstream/${ branch }`, true, "gitMergeUpstreamBranch" );
+}
+
+export function gitMergeUpstreamMaster( state ) {
+	return git.mergeUpstreamMaster();
+}
+
+export function gitMergeUpstreamDevelop( state ) {
+	const { hasDevelopBranch } = state;
+
+	return hasDevelopBranch ? git.mergeUpstreamDevelop() : Promise.resolve();
+}
+
+export function checkHasDevelopBranch( state ) {
+	return git.getRemoteBranches().then( data => {
+		const branches = data.split( "\n" );
+		state.hasDevelopBranch = branches.some( branch => ~branch.trim().indexOf( "upstream/develop" ) );
+	} ).catch( () => {
+		state.hasDevelopBranch = false;
+	} );
+}
+
+export function askPrereleaseIdentifier( state ) {
+	const { identifier } = state;
+
+	if ( identifier && identifier.length ) {
+		return Promise.resolve();
+	}
+
+	return util.prompt( [ {
+		type: "input",
+		name: "prereleaseIdentifier",
+		message: "Pre-release Identifier:"
+	} ] ).then( response => {
+		state.identifier = response.prereleaseIdentifier;
+		return Promise.resolve();
+	} );
+}
+
+export function gitCheckoutMaster( state ) {
+	state.branch = "master";
+	return git.checkoutMaster();
+}
+
+export function getCurrentBranchVersion( state ) {
+	const { configPath } = state;
+
+	let pkg = {};
+	try {
+		pkg = util.readJSONFile( configPath );
+	} catch ( err ) {
+		util.advise( "updateVersion" );
+	}
+
+	state.currentVersion = pkg.version;
+	return Promise.resolve();
+}
+
+export function gitShortLog( state ) {
+	const { currentVersion, prerelease } = state;
+
+	let contents = util.readFile( CHANGELOG_PATH );
+
+	if ( ~contents.indexOf( "### Next" ) ) {
+		contents = contents.replace( /### Next([^#]+)/, ( match, submatch ) => {
+			state.log = submatch.trim();
+			return "";
+		} );
+
+		util.writeFile( CHANGELOG_PATH, contents );
+	} else {
+		return git.getTagList().then( tags => {
+			let latestRelease = `v${ currentVersion }`;
+			if ( tags.length ) {
+				if ( !prerelease ) {
+					tags = tags.filter( tag => !tag.includes( "-" ) );
+					latestRelease = tags[ tags.length - 1 ];
+				}
+			}
+
+			return git.shortLog( latestRelease ).then( data => {
+				data = data.trim().replace( /^(.+)$/gm, "* $1" );
+				if ( !data.length ) {
+					util.advise( "gitLog.log", { exit: false } );
+				}
+
+				state.log = data;
+			} );
+		} ).catch( () => util.advise( "gitLog.log" ) );
+	}
+}
+
+export function previewLog( { log } ) {
+	logger.log( `${ chalk.bold( "Here is a preview of your log:" ) }
+${ chalk.green( log ) }` );
+}
+
+export function askSemverJump( state ) {
+	const { currentVersion, identifier, prerelease, release } = state;
+
+	// don't bother prompting if this information was already provided in the CLI options
+	if ( release && release.length ) {
```

Comment:
i wonder if checks like this should be done before this point... like when the steps to be used are determined? (dynamically removing them from the array of steps)

if this is too involved, maybe it could be a future refactor